### ISSUE TITLE
Unable to use `sui-faucet` on local sui network that is started using `sui genesis-ceremony`

### ISSUE THREAD

**Issue by KevL-99** [MystenLabs/sui#16420]
# Goal
After starting a local sui network using `sui genesis-ceremony`, start `sui-faucet` and get some gas coins

# Issue
1. Lack of documentation on running `sui-faucet` on a local network started using `sui genesis-ceremony`
2. `sui faucet` returns an error when there are 0 gas coins 
3. After an ad-hoc modification of `simple_faucet.rs`, `sui-faucet` runs but unable to request gas coins

# Steps to reproduce the issue
1. Start up fullnodes and validators by running `cargo run --bin sui-node -- --config-path <path/to/validator/or/fullnode/config/files>`
2. Run `sui client gas`, there are no gas coins owned by the active address
3. run `cargo run --bin sui-faucet -- --write-ahead-log /tmp/faucet.wal`, the faucet will give error: `mpsc bounded channel requires buffer > 0`, the problem is caused by lines 124 and 125 of `simple_faucet.rs`, containing `mpsc::channel(coins.len())`, but there are no coins! So I manually changed it to `mpsc::channel(100)`, the faucet then starts up
4. make a request to the faucet:
```
    curl --location --request POST <local faucet address> \
    --header 'Content-Type: application/json' \
    --data-raw '{
        "FixedAmountRequest": {
            "recipient": "<active sui address>"
        }
    }'
```
5. Faucet log gives error
```
ERROR sui_faucet::faucet::simple_faucet: Timeout when getting gas coin from the queue uuid=927d9127-e430-4da9-9f8c-f150c214dd34
WARN sui_faucet::faucet::simple_faucet: Failed getting gas coin, try later!
WARN sui_faucet: Failed to request gas: NoGasCoinAvailable uuid=927d9127-e430-4da9-9f8c-f150c214dd34
WARN sui_faucet::metrics_layer: Request failed in 5.01s, code: 500
```
6. `curl` command gives error response  `{"transferredGasObjects":[],"error":"Timed out waiting for a coin from the gas coin pool"}`
7. If I do`sui genesis -f` and `sui start`, the start the faucet, the addresses seem to have some gas coin right off the start, so I was able to start the faucet without changing `mpsc::channel(coins.len())`, and the `curl` command gives gas coin to the active address

**Comment by KunPengRen** [comment]
hi, did you start the faucet service successfully?

**Comment by KevL-99** [comment]
> hi, did you start the faucet service successfully?

No. However I was able to use `sui genesis -f --benchmark-ips` to start a local network with an arbitrary number of machines/IP addresses, and the faucet was usable with this method.

**Comment by b0xtch** [comment]
I am facing the same issue now running the faucet were there any updates on how to run the faucet locally?

**Comment by KevL-99** [comment]
> I am facing the same issue now running the faucet were there any updates on how to run the faucet locally?
I ended up using `sui genesis -f --benchmark-ips` instead of `sui genesis-ceremony` to start a local network with an arbitrary number of machines/IP addresses, they do the same thing. The faucet works with `sui genesis -f --benchmark-ips`

**Comment by b0xtch** [comment]
> 3\. cargo run --bin sui-faucet -- --write-ahead-log /tmp/faucet.wal

awesome thanks, do you just run the faucet with

```
cargo run --bin sui-faucet -- --write-ahead-log /tmp/faucet.wal
```

**Comment by b0xtch** [comment]
I am running

```
sui genesis -f --benchmark-ips
```

but that just clear the genesis do you run this first then `sui start`?

**Comment by KevL-99** [comment]
> > 3. cargo run --bin sui-faucet -- --write-ahead-log /tmp/faucet.wal
> 
> awesome thanks, do you just run the faucet with
> 
> ```
> cargo run --bin sui-faucet -- --write-ahead-log /tmp/faucet.wal
> ```
> 
> @KevL-99

yes

**Comment by KevL-99** [comment]
> I am running
> 
> ```
> sui genesis -f --benchmark-ips
> ```
> 
> but that just clears the genesis do you run this first then `sui start`?

yes after genesis you will need to do sui start, or if you are running a network on multiple machines, do `cargo run --bin sui-node -- --config-path <path/to/validator/or/fullnode/config/files>`, the config files are generated by `sui-genesis`

**Comment by b0xtch** [comment]
> > I am running
> > ```
> > sui genesis -f --benchmark-ips
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > but that just clears the genesis do you run this first then `sui start`?
> 
> yes after genesis you will need to do sui start, or if you are running a network on multiple machines, do `cargo run --bin sui-node -- --config-path <path/to/validator/or/fullnode/config/files>`, the config files are generated by `sui-genesis`

beauty thank you
